src/main/java/frc/robot/subsystems/drive/DriveSubsystem.java

```diff
@@ -26,8 +46,36 @@ public DriveSubsystem(ChassisMode chassisMode, DriveMode driveMode) {
             driveHardware = new DriveVictor(DriveConstants.frontLeftID, DriveConstants.frontRightID, DriveConstants.backLeftID, DriveConstants.backRightID);
         }
         mode = driveMode;
+
+        driveModules[0] = new SwerveModules(DriveConstants.frontLeftID, DriveConstants.frontLeftRotationMotorID,
+        DriveConstants.frontLeftRotationEncoderID, DriveConstants.frontLeftAngleOffset, false, false);
+        driveModules[1] = new SwerveModules(DriveConstants.frontRightID, DriveConstants.frontRightRotationMotorID,
+        DriveConstants.frontRightRotationEncoderID, DriveConstants.frontRightAngleOffset, true, false);
+        driveModules[2] = new SwerveModules(DriveConstants.backLeftID, DriveConstants.backLeftRotationMotorID,
+        DriveConstants.backLeftRotationEncoderID, DriveConstants.backLeftAngleOffset, false, false);
+        driveModules[3] = new SwerveModules(DriveConstants.backRightID, DriveConstants.backRightRotationMotorID,
+        DriveConstants.backRightRotationEncoderID, DriveConstants.backRightAngleOffset, true, false);
+
+        for (int i = 0; i < 4; i++) {
+            driveModules[i].zeroEncoders();
+            modulePositions[i] = new SwerveModulePosition();
+            goalModuleStates[i] = new SwerveModuleState();
+
+            rotationPIDs[i] = new PIDController(DriveConstants.rotationKps[i], 0, DriveConstants.rotationKds[i]);
+            rotationPIDs[i].enableContinuousInput(-Math.PI, Math.PI);
```

Comment:
Let's make the rotationPIDs a part of the SwerveModules class, since it seems like that class should be able to manage its own rotation control

Reply:
done